### PR TITLE
Update the integration for the compatibility with pre-1.0.0 version of the Neptune client

### PR DESCRIPTION
**Pull request by twolodzko**

### CODE REVIEW

**Review by AleksanderWWW — COMMENTED** [review]
@twolodzko left two questions; apart from that looks good

File: src/neptune_optuna/impl/__init__.py [hunk 1]
```diff
@@ -144,8 +146,8 @@ def __init__(
     ):
 
         expect_not_an_experiment(run)
-        verify_type("run", run, neptune.Run)
-        verify_type("base_namespace", base_namespace, str)
+        verify_type("run", run, (neptune.Run, neptune.handler.Handler))
+        verify_type("base_namespace", base_namespace, (str, type(None)))
```
[the remark this answers is not in the document]
Comment (twolodzko) [reply src/neptune_optuna/impl/__init__.py:150]: It is, but now the default is `None` for a more explicit empty case.

File: src/neptune_optuna/impl/__init__.py [hunk 2]
```diff
@@ -171,7 +173,11 @@ def __init__(
         )
         verify_type("target_names", target_names, (list, type(None)))
 
-        self.run = run[base_namespace]
+        if base_namespace is not None:
```
[the remark this answers is not in the document]
Comment (twolodzko) [reply src/neptune_optuna/impl/__init__.py:176]: The default is None. What SyntaxWarning do you mean?

[the remark this answers is not in the document]
Comment (AleksanderWWW) [reply src/neptune_optuna/impl/__init__.py:176]: Oh, sorry for confusion, yes, you're right.

**Review by AleksanderWWW — APPROVED** [review]

File: src/neptune_optuna/impl/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (twolodzko) [reply src/neptune_optuna/impl/__init__.py]: OK, reverting the change in base_namespace, it's inconsistent with other integrations.

**Pull request merged**